### PR TITLE
{anytype-legacy,anytype}: 0.31.0 -> 0.32.0

### PR DESCRIPTION
**Pull request by bbigras**
- anytype: 0.31.0 -> 0.32.0
- anytype-legacy: init at 0.31.8

0.31.8 is the last version using the old data format or whatever.

Users wanting to migrate needs to export their data with the legacy version.

migration guide:
https://community.anytype.io/t/anytype-legacy-to-beta-migration-trail-guide/9274

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/anytype/legacy.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
+{ lib, fetchurl, appimageTools }:
+
+let
+  pname = "anytype";
+  version = "0.31.8";
+  name = "Anytype-${version}";
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/anytype/legacy.nix:6]:
```suggestion
```

File: pkgs/applications/misc/anytype/legacy.nix [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  nameExecutable = pname;
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/anytype/legacy.nix:7]:
```suggestion
```

File: pkgs/applications/misc/anytype/legacy.nix [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  src = fetchurl {
+    url = "https://at9412003.fra1.digitaloceanspaces.com/Anytype-${version}.AppImage";
+    name = "Anytype-${version}.AppImage";
+    sha256 = "sha256-Ypz+tPUuqj+3wjWo/0eyTZ8oK+HffByEe+IYRQzCdUM=";
+  };
+  appimageContents = appimageTools.extractType2 { inherit name src; };
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/anytype/legacy.nix:13]:
```suggestion
  appimageContents = appimageTools.extractType2 { inherit pname version src; };
```

File: pkgs/applications/misc/anytype/legacy.nix [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+in
+appimageTools.wrapType2 {
+  inherit name src;
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/anytype/legacy.nix:16]:
```suggestion
  inherit pname version src;
```

File: pkgs/applications/misc/anytype/legacy.nix [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+  extraPkgs = pkgs: (appimageTools.defaultFhsEnvArgs.multiPkgs pkgs)
+    ++ [ pkgs.libsecret ];
+
+  extraInstallCommands = ''
+    mv $out/bin/${name} $out/bin/${pname}
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/anytype/legacy.nix:22]:
```suggestion
    mv $out/bin/${pname}-${version} $out/bin/${pname}
```

File: pkgs/applications/misc/anytype/default.nix [hunk 6]
```diff
@@ -2,13 +2,13 @@
 
 let
   pname = "anytype";
-  version = "0.31.0";
+  version = "0.32.0";
   name = "Anytype-${version}";
   nameExecutable = pname;
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/anytype/default.nix:7]:
```suggestion
```
same as below

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/anytype/default.nix [hunk 7]
```diff
@@ -2,13 +2,11 @@
...
-  nameExecutable = pname;
+  version = "0.32.0";
   src = fetchurl {
-    url = "https://at9412003.fra1.digitaloceanspaces.com/Anytype-${version}.AppImage";
+    url = "https://anytype-release.fra1.digitaloceanspaces.com/Anytype-${version}.AppImage";
     name = "Anytype-${version}.AppImage";
-    sha256 = "sha256-s8al0R9G478A+PymQcdcdRpw6tpKkG+WIZsXZYEvf/o=";
+    sha256 = "sha256-OzYSDYh/bGrwacajtA3+GHZczMYsF3y8cQEPTeoyplM=";
   };
   appimageContents = appimageTools.extractType2 { inherit name src; };
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/anytype/default.nix:11]:
```suggestion
  appimageContents = appimageTools.extractType2 { inherit pname src; };
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by bbigras** [comment]
@SuperSandro2000 thank you very much, as always, for your help, but I think I'll close this PR.

People wanting to use the old version for migration could use [Nixhub.io](https://www.nixhub.io/) or https://lazamar.co.uk/nix-versions/

**Pull request closed without merging**